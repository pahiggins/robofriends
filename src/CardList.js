import React, { Component } from 'react';
import Card from './Card';

class CardList extends Component {
    render() {
        return this.props.robots.map(robot => <Card key={robot.id} name={robot.name} email={robot.email} />);
    }
}

export default CardList;