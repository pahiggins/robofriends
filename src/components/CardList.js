import React from 'react';
import Card from './Card';

const CardList = (props) => {
    return props.robots.map(robot => <Card key={robot.id} name={robot.name} email={robot.email} />);
}

export default CardList;