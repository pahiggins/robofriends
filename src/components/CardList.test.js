import { shallow } from 'enzyme';
import React from 'react';
import CardList from './CardList';

it('expect to render CardList component', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'Peter Higgins',
            email: 'pah@test.com'
        }
    ];
    expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});