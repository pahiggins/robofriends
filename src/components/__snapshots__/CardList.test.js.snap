// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
        Array [
            Object {
              "email": "pah@test.com",
              "id": 1,
              "name": "Peter Higgins",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <Card
      email="pah@test.com"
      id={1}
      name="Peter Higgins"
/>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <Card
        email="pah@test.com"
        id={1}
        name="Peter Higgins"
/>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
